### PR TITLE
Gisaid region fields

### PR DESCRIPTION
**Pull request by mayabrandi**
## Description
*This PR makes the gisaiud upload read new lims sample fields to generate the summary scv file.*

### How to prepare for test
- [ ] ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] paxa the environment: `paxa`
- [ ] run:
```
bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-servers-stage.sh master
bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-cg-stage.sh gisaid-region-fields
/home/proj/stage/servers/resources/gisaid_uploader.py CoV authenticate --cid <client id>
cg -c /home/proj/stage/servers/config/hasta.scilifelab.se/cg-stage.yaml upload gisaid frankhusky
```

## Review
- [ ] code approved by
- [x] tests executed by @mayabrandi 
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan
- [ ] Document in ...
- [ ] Deploy this branch
- [ ] Inform to ...

### CODE REVIEW

**Comment by mayabrandi** [comment]
Testing on hasta.

```

[0|0|510] [maya.brandi@hasta:~] [S_main] $ cg -c /home/proj/stage/servers/config/hasta.scilifelab.se/cg-stage.yaml upload gisaid frankhusky
/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))
----------------- UPLOAD ----------------------
----------------- GISAID -------------------
/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Demux.samples to refer to viewonly relationship Sample.demuxes should include sync_backref=False set on the Demux.samples relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Sample.demuxes to refer to viewonly relationship Demux.samples should include sync_backref=False set on the Sample.demuxes relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
Fetch latest version from bundle frankhusky
Fetching files with tags in [consensus]
Fetching files from version 40010
Running command /home/proj/stage/servers/resources/gisaid_uploader.py CoV upload --csv /home/maya.brandi/frankhusky.csv --fasta /home/maya.brandi/frankhusky.fasta
gisaid stdout:
hCoV-19/Sweden/01_SE100_21CS500657TEST/2020-10-31; EPI_ISL_2238708
hCoV-19/Sweden/01_SE100_21CS500658TEST/2020-11-04; EPI_ISL_2238709
hCoV-19/Sweden/01_SE100_21CS500659TEST/2020-11-06; EPI_ISL_2238710
hCoV-19/Sweden/01_SE100_21CS500660TEST/2020-11-08; EPI_ISL_2238711
hCoV-19/Sweden/01_SE100_21CS500661TEST/2020-11-10; EPI_ISL_2238712
hCoV-19/Sweden/01_SE100_21CS500662TEST/2020-11-11; EPI_ISL_2238713
hCoV-19/Sweden/01_SE100_21CS500663TEST/2020-11-12; EPI_ISL_2238714
hCoV-19/Sweden/01_SE100_21CS500664TEST/2020-11-13; EPI_ISL_2238715
hCoV-19/Sweden/01_SE100_21CS500665TEST/2020-11-03; EPI_ISL_2238716
hCoV-19/Sweden/01_SE100_21CS500666TEST/2020-11-08; EPI_ISL_2238717
hCoV-19/Sweden/01_SE100_21CS500667TEST/2020-11-07; EPI_ISL_2238718
hCoV-19/Sweden/01_SE100_21CS500668TEST/2020-11-09; EPI_ISL_2238719
submissions uploaded: 12
submissions failed: 0
```
👍

File: cg/meta/upload/gisaid/gisaid.py [hunk 1]
```diff
@@ -23,49 +24,55 @@ class GisaidAPI:
...
-            ).first()
-            fasta_sequence: str = self.get_fasta_sequence(fastq_path=fasta_file.full_path)
-            fasta_obj = FastaFile(header=sample.covv_virus_name, sequence=fasta_sequence)
-            fasta_objects.append(fasta_obj)
-        return fasta_objects
-
-    def build_gisaid_fasta(self, gsaid_samples: List[GisaidSample], file_name: str) -> Path:
+        hk_version: hk_models.Version = self.housekeeper_api.last_version(bundle=family_id)
+        if not hk_version:
+            LOG.info("Family ID: %s not found in hose keeper", family_id)
```
Comment (Mropat) [inline cg/meta/upload/gisaid/gisaid.py:47]:
```suggestion
            LOG.info("Family ID: %s not found in housekeeper", family_id)
```

File: cg/meta/upload/gisaid/gisaid.py [hunk 2]
```diff
@@ -101,14 +108,26 @@ def get_gsaid_samples(self, family_id: str) -> List[GisaidSample]:
         samples: List[models.Sample] = self.status_db.get_sequenced_samples(family_id=family_id)
         gisaid_samples = []
         for sample in samples:
+            sample_id = sample.internal_id
```
Comment (Mropat) [inline cg/meta/upload/gisaid/gisaid.py:111]:
```suggestion
            sample_id: str = sample.internal_id
```

File: cg/meta/upload/gisaid/models.py [hunk 3]
```diff
@@ -15,20 +16,17 @@ class UpploadFiles(BaseModel):
     fasta_file: Path
 
 
-class Lab(BaseModel):
-    city: str
-    address: str
-    region_nr: str
-    institute: str
+#    class Config:
```
Comment (Mropat) [inline cg/meta/upload/gisaid/models.py:19]:
```suggestion
```

File: cg/meta/upload/gisaid/models.py [hunk 4]
```diff
@@ -45,43 +43,15 @@ class GisaidSample(BaseModel):
...
-        return lab.institute
+        region: str = values.get("region")
+        return f"Europe/Sweden/{region}"
 
     @validator("covv_subm_sample_id", always=True)
     def parse_subm_sample_id(cls, v, values):
-        lab = values.get("lab")
-        return f"{lab.region_nr}_SE100_{v}"
+        region_code = values.get("region_code")
+        return f"{region_code}_SE100_{v}"
```
Comment (Mropat) [inline cg/meta/upload/gisaid/models.py:54]: Is the SE100 not lab code? in that case it is not always SE100 and can also be fetched from lims

Comment (mayabrandi) [reply cg/meta/upload/gisaid/models.py:54]: According to Robert it should allways be SE100. I can double check. with him

Comment (mayabrandi) [reply cg/meta/upload/gisaid/models.py:53]: Robert comfirmed: Allways SE100

**Review by Mropat — APPROVED** [review]

**Pull request merged**